x/ccv/provider/keeper/grpc_query.go

```diff
@@ -437,29 +437,27 @@ func (k Keeper) hasToValidate(
 	if err != nil {
 		return false, nil
 	}
+
+	minPowerToOptIn := int64(0)
+	// If the consumer is TopN compute the minimum power
 	if topN := k.GetTopN(ctx, consumerId); topN > 0 {
-		// in a Top-N chain, we automatically opt in all validators that belong to the top N
-		minPower, found := k.GetMinimumPowerInTopN(ctx, consumerId)
-		if found {
-			k.OptInTopNValidators(ctx, consumerId, activeValidators, minPower)
-		} else {
-			k.Logger(ctx).Error("did not find min power in top N for chain", "chain", consumerId)
+		minPowerToOptIn, err = k.ComputeMinPowerInTopN(ctx, activeValidators, topN)
```

Comment:
Why do we use `ComputeMinPowerInTopN` instead of `GetMinimimumPowerInTopN` here?
I thought we wanted to use `GetMinimimumPowerInTopN` because it's computationally cheap?

Reply:
Both `ComputeMinPowerInTopN` and `GetMinimimumPowerInTopN` aren't updating the sate and thus can be used in queries.
We use the former since we the query shoud use the current min power to opt-In.